api/environments/views.py

```diff
@@ -140,6 +140,12 @@ def get_queryset(self):
         return queryset
 
     def perform_create(self, serializer):
+        existing_environment = Environment.objects.filter(
+            name=serializer.validated_data["name"],
+            project=serializer.validated_data["project"],
+        )
+        if existing_environment:
+            raise ValidationError("Existing environment for given name.")
```

Comment:
I think we can do this in a much simpler way using the serializer as per the PR I linked on the issue (see [here](https://github.com/Flagsmith/flagsmith/pull/5039))

Reply:
Sorry I missed the linked PR! I've updated the code to handle this in a similar manner to what you posted.